### PR TITLE
Migrated User Manual from Docs to Sphinx

### PR DESCRIPTION
**Pull request by rorymcclenahan**
## Description
- Migrate the user manual to the version control and is now built using Sphinx instead of Docs. The manual is now built using Sphinx to automatically generate static web pages from restructured text (rst).
- Created a Sphinx role extension, called only, to help differentiate between the different versions of XDMoD.
- Remove existing portal_settings.ini reference to user manual and replaced with relative link in CCR.js
User Manual Home Page

## Motivation and Context
Migrating the user manual will allow us to include the user manual under the version control and simplify the build process to generate the user manual. It can now be edited directly on GitHub, and it reduces maintainability of the user manual.

## Tests performed
All the automated tests for a fresh install of rocky Linux 8
Created tests to ensure that each chapter had an associated html file after build, and that XSEDE/ACCESS chapters were not showing up when generating the Open version.

## Checklist:

- [x] The pull request description is suitable for a Changelog entry
- [x] The milestone is set correctly on the pull request
- [x] The appropriate labels have been added to the pull request

### CODE REVIEW

File: user_manual_builder/conf.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'XDMoD Manual'
+copyright = '2023, UB CCR'
+author = 'UB CCR'
+release = '11.0'
```
Comment (jpwhite4) [inline user_manual_builder/conf.py:16]: Don't hard code this version number - instead it should be obtained from the contents of the build.json file (which is the autoritative source.

Comment (jpwhite4) [inline user_manual_builder/conf.py:16]: Don't hard code this version number - instead it should be obtained from the contents of the build.json file (which is the autoritative source).

File: user_manual_builder/conf.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+numfig = True
+
+
+# -- Options for HTML output -------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
+
+html_theme = 'sphinx_rtd_theme'
+html_static_path = ['_static']
+html_logo = 'media/image43.png'
```
Comment (jpwhite4) [inline user_manual_builder/conf.py:34]: Please can you change this image name to something that describes the image content. For example, if this is the XDMoD logo then change the name to `xdmod_logo.png`

**Review by jpwhite4 — CHANGES_REQUESTED** [review]
You changed the portal_settings.ini configuration options. You also need to update the documentation about the portal_settings configuration options in the docs/configuration.md file.

File: user_manual_builder/conf.py (hunk 1 above)
Comment (jpwhite4) [reply user_manual_builder/conf.py:16]: Similarly, I suggest dynamically generating the year in the copyright rather than hardcoding

**Comment by jpwhite4** [comment]
Please remove the "edit in github" link.

**Comment by jpwhite4** [comment]
Please can you also add a test in the validate script (https://github.com/ubccr/xdmod/blob/xdmod10.5/tests/ci/validate.sh) that confirms that the manual was installed correctly.

File: tests/ci/bootstrap.sh [hunk 3]
```diff
@@ -29,7 +29,7 @@ fi
 
 # Install python dependencies for the image hash comparison algorithm
 yum install -y python3 python3-six python3-numpy python3-pillow ${PYTHON_SCIPY}
-pip3 install imagehash==4.2.1
+python3 -m pip install imagehash==4.2.1
```
Comment (jpwhite4) [inline tests/ci/bootstrap.sh:32]:
```suggestion
pip3 install imagehash==4.2.1
```

File: tests/regression/lib/Controllers/UsageChartsTest.php [hunk 4]
```diff
@@ -114,7 +114,8 @@ private function phash($type, $imageData)
             fclose($pipes[2]);
             $retval = proc_close($process);
             if (strlen($err) > 0 || $retval !== 0) {
-                throw new Exception("imagehash returned $retval stderr='$err'");
+                throw new \Exception("imagehash returned $retval stderr='$err'");
```
Comment (jpwhite4) [inline tests/regression/lib/Controllers/UsageChartsTest.php:117]: Prefer this to be in a different pull request.

**Review by jpwhite4 — CHANGES_REQUESTED** [review]

File: html/gui/js/CCR.js [hunk 5]
```diff
@@ -377,13 +377,13 @@ XDMoD.GlobalToolbar.Help = function (tabPanel) {
             iconCls: 'user_manual_16',
             id: 'global-toolbar-help-user-manual',
             handler: function () {
-                if (tabPanel === undefined) {
-                    window.open("user_manual.php");
+                const userManualSectionName = tabPanel.getActiveTab().userManualSectionName;
```
Comment (jpwhite4) [inline html/gui/js/CCR.js:380]: Need to check if tabPanel is defined before dereferencing it.

**Review by jpwhite4 — CHANGES_REQUESTED** [review]

File: html/gui/js/CCR.js (hunk 5 above)
Comment (connersaeli) [reply html/gui/js/CCR.js]: See `html/gui/js/CCR.js` for resolution

File: tests/ci/bootstrap.sh (hunk 3 above)
Comment (connersaeli) [reply tests/ci/bootstrap.sh]: https://github.com/ubccr/xdmod/pull/1744/commits/cac9280eb42f5095fef229ad788f1c8df55b1dd3

File: tests/regression/lib/Controllers/UsageChartsTest.php (hunk 4 above)
Comment (connersaeli) [reply tests/regression/lib/Controllers/UsageChartsTest.php]: Reverted: https://github.com/ubccr/xdmod/pull/1744/commits/cac9280eb42f5095fef229ad788f1c8df55b1dd3
Separate PR: https://github.com/ubccr/xdmod/pull/1819

**Pull request merged**